File: sol/solution/8 Kyu/56efc695740d30f963000557/Solution.ts
/*
Solution
*/
export function toAlternatingCase(s: string): string {
  return s
    .split("")
    .map(ch => ch === ch.toLowerCase() ? ch.toUpperCase() : ch.toLowerCase())
    .join("");
}





/*
Sample Tests
*/
import solution = require('./solution');
// import the type of assertion library you wish to use (Chai recommended)
import {assert} from "chai";

describe("Sample Test Cases", function(){
  it("Should return a revesed case value", function() {
    assert.equal(solution.toAlternatingCase("hello world"), "HELLO WORLD");
    assert.equal(solution.toAlternatingCase("HeLLo WoRLD"), "hEllO wOrld");
    assert.equal(solution.toAlternatingCase("1a2b3c4d5e"), "1A2B3C4D5E");
  });
});

File: sol/solution/8 Kyu/56efc695740d30f963000557/Solution.hs
{-
Solution
-}
module Codewars.Kata.AlternatingCase where
import Data.Char (isLower, isUpper, toLower, toUpper)
toAlternatingCase :: String -> String
toAlternatingCase = map swapCase
  where
    swapCase c
      | isLower c = toUpper c
      | isUpper c = toLower c
      | otherwise = c





{-
Sample Tests
-}
module Codewars.Kata.AlternatingCaseSpec (spec, main) where
import Codewars.Kata.AlternatingCase (toAlternatingCase)
import Codewars.Kata.AlternatingCase.ASCII
import Test.Hspec
import Test.QuickCheck

main = hspec spec
spec = describe "toAlternatingCase" $ do
  it "works for the examples" $ do
    toAlternatingCase "hello world" `shouldBe` "HELLO WORLD"
    toAlternatingCase "HELLO WORLD" `shouldBe` "hello world"
    toAlternatingCase "hello WORLD" `shouldBe` "HELLO world"
    toAlternatingCase "HeLLo WoRLD" `shouldBe` "hEllO wOrld"
    toAlternatingCase "12345"       `shouldBe` "12345"
    toAlternatingCase "1a2b3c4d5e"  `shouldBe` "1A2B3C4D5E"
  
  it "returns the same string if called twice" $ property $ \(ASCII xs) ->
    toAlternatingCase (toAlternatingCase xs) `shouldBe` xs
  
  it "returns a string with the same length" $ property $ \(ASCII xs) ->
    let l = length xs
    in toAlternatingCase xs `shouldSatisfy` ((l ==) . length)